lib/streamlit/elements/widgets/time_widgets.py

```diff
@@ -571,7 +571,7 @@ def _time_input(
     def date_input(
         self,
         label: str,
-        value: NullableScalarDateValue | None = "today",
+        value: DateValue | None = "today",
```

Comment:
I may be missing something here but I think that `DateValue` already accounts for the fact that `value` can be `None`. Please see [here](https://github.com/streamlit/streamlit/blob/1.41.0/lib/streamlit/elements/widgets/time_widgets.py#L71)